elastic/kibana
# [ObsUX] Enable Hosts advanced setting on serverless

Closes https://github.com/elastic/kibana/issues/169198

## Summary

* Adds Hosts advanced setting to the observability settings allowlist
* Adds tests for Hosts sidenav item visibility

https://github.com/elastic/kibana/assets/793851/3c9252c8-a4cb-4da8-aa60-23a8aeb60d4c

# How to test

* Run locally in serverless mode
* Make sure "Observability Hosts view" toggle is visible on the Advanced Settings screen
* Toggle the setting on and off, make sure sidenav properly reacts to the changes
* Go to Infra settings screen
* Toggle the same setting on and off, and also make sure sidenav properly reacts to the changes
